tests/unit/test_datablock.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Apache License, Version 2.0,
+ * modified with the Commons Clause restriction.
+ */
+
+#include "../../deps/googletest/include/gtest/gtest.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdio.h>
+#include <string.h>
+#include "../../src/util/datablock/datablock.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+class DataBlockTest: public ::testing::Test {
+};
+
+
+/*
+void DataBlock_MigrateItem(DataBlock *dataBlock, size_t src, size_t dest)
+void DataBlock_Free(DataBlock *block)
+*/
+TEST_F(DataBlockTest, NewDataBlock) {
+    // Create a new data block, which can hold 1024 items
+    // each item is an integer.
+    size_t itemSize = sizeof(int);
+    DataBlock *dataBlock = DataBlock_New(1024, itemSize);
+
+    EXPECT_EQ(dataBlock->itemCount, 0);     // No items were added. 
+    EXPECT_EQ(dataBlock->itemCap, 1024);
```

Comment:
Fails because `itemCap` will always be a multiple of `BLOCK_CAP`, which for us is 16,384 - `datablock.c:74`